charge/servers/molecular_property_utils.py

```diff
@@ -0,0 +1,58 @@
+################################################################################
+## Copyright 2025 Lawrence Livermore National Security, LLC. and Binghamton University.
+## See the top-level LICENSE file for details.
+##
+## SPDX-License-Identifier: Apache-2.0
+################################################################################
+
+try:
+    from rdkit import Chem
+    from rdkit.Chem import AllChem, Descriptors
+    from rdkit.Contrib.SA_Score import sascorer
+except ImportError:
+    raise ImportError("Please install the rdkit package to use this module.")
+
+from loguru import logger
+from charge.servers.SMILES_utils import get_synthesizability
+
+def get_density(smiles: str) -> float:
+    """
+    Calculate the density of a molecule given its SMILES string.
```

Comment:
No units, not enough description of what density is

Reply:
Done.